nixos/modules/programs/ssh.nix

```diff
@@ -27,7 +27,7 @@ let
     )) + "\n";
 
   knownHostsFiles = [ "/etc/ssh/ssh_known_hosts" ]
-    ++ map pkgs.copyPathToStore cfg.knownHostsFiles;
+    ++ map (f: "${f}") cfg.knownHostsFiles;
```

Comment:
This is a breaking change:

```
nix-repl> builtins.filterSource (p: t: true) "/etc/os-release"  
"/nix/store/y3nska2hgd2r0iqjh9yvxl1d920y90ay-os-release"

nix-repl> "${"/etc/os-release"}"                               
"/etc/os-release"

```

Arguably this breaks in the right direction, but I would still suggest to introduce a new option for the new semantics, just to be sure: `cfg.knownHostsPaths`?
(_Paths_ suggest that they can be just that, and not _file_ contents, so I think that's nice. Not very precise but still. I guess what matters is that it's different, fwiw)

Reply:
Hmm, right. The proposed behaviour is in line with all other nixos options though, where we have the escape hatch of passing a path in a string when the file should not be copied to the nix store and we really want the given path to be included verbatim.

This is the only place in nixos where `copyPathToStore` is used, nowhere else do we have the same semantics.

So yeah, the question is whether we want to clean this up, or instead introduce a precedent of having two different options for including files with different behaviour.

The proposed option is strictly more powerful since you can still manually call `copyPathToStore` if that's really what you want (but it's only useful for absolute paths, which should be discouraged anyway).
For the usual use-case of passing either relative paths or derivations, the behaviour stays the same (minus the unnecessary IFD).

Reply:
I don't see why you'd ever want to copy some file in /etc/ to the Nix Store to include as part of a NixOS closure. That's just ...wrong.

I'd say leave a release note on this and point anyone relying on that behaviour to copy their known hosts to their config dir and include it from there instead.

Reply:
Alright, @roberth I introduced the new option as proposed and added a deprecation warning for the old one so that we can hopefully remove it at some point.